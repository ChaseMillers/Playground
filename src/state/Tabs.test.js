import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer'
import Tabs from './Tabs';
import { shallow } from 'enzyme'

describe(`Tabs Component`, () => {

    const tabsProp = [
        { name: '1st tab',
          content: <p>People</p> },
        { name: '2nd tab',
          content: <p>Order</p> },
        { name: '3rd tab',
          content: <p>Our</p> },
        { name: '4th tab',
          content: <p>Potatoes</p> }
      ];

  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Tabs />, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the first tab by default', () => {
    const tree = renderer.create(<Tabs tabs={tabsProp} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders empty given no tabs', () => {
    const wrapper = shallow(<Tabs />)
    })
})

