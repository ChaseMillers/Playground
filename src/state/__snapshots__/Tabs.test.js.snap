// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs Component renders the first tab by default 1`] = `
<div>
  <hr />
  <button>
    1st tab
  </button>
  <button>
    2nd tab
  </button>
  <button>
    3rd tab
  </button>
  <button>
    4th tab
  </button>
  <div
    className="content"
  >
    <p>
      People
    </p>
  </div>
  <hr />
</div>
`;
